code_generation/license.cpp

```diff
@@ -136,3 +137,32 @@ QString License::text() const
 
     return txt;
 }
+
+License::Type License::getLicenseNameByString(const QString &license_string)
+{
+    bool result;
+    QMetaEnum metaEnum = QMetaEnum::fromType<License::Type>();
+    License::Type license_enum = static_cast<License::Type>(
+            metaEnum.keyToValue(license_string.toStdString().c_str(), &result));
+    if (result) {
+        return license_enum;
+    } else {
+        return License::Type::NoLicense;
+    }
+}
+
+QString License::getLicenseNameByEnum(const License::Type &license_enum)
+{
+    QMetaEnum metaEnum = QMetaEnum::fromType<License::Type>();
+    return metaEnum.valueToKey(license_enum);
+}
+
+QString License::getSupportedLicenses()
+{
+    QString result;
+    QMetaEnum metaEnum = QMetaEnum::fromType<License::Type>();
+    for (int i = 0; i < metaEnum.keyCount(); i++) {
+        result = result + metaEnum.valueToKey(i) + " ";
```

Comment:
`result += `

QLatin1Char(' '), no need for a null-terminated string

Isn't it a problem to have a trailing space after the last item?